torch/fx/experimental/sym_node.py

```diff
@@ -128,7 +128,8 @@ def with_shape_env(self, shape_env: "ShapeEnv") -> "SymNode":
 
     @property
     def expr(self):
-        return self.shape_env.replace(self._expr)
+        # NB: must NOT resolve unbacked SymInts
+        return self.shape_env.replace(self._expr, resolve_unbacked=False)
```

Comment:
Is there a guarantee that unbacked symints are never replaced by other paths? Otherwise worst case this call will do nothing...

Reply:
I think _expr is immutable and we never modify it. Which I suppose suggests a simpler implementation plan where we just have an alternate accessor that gives you the base symbol without any replacements

Reply:
You mean just returning `_expr` itself?

Reply:
Yup lol